monitor/checks.go

```diff
@@ -33,6 +37,71 @@ func checkPaths(config *keysync.Config) []error {
 	return errs
 }
 
+func checkClientHealth(config *keysync.Config) []error {
+	clients, err := config.LoadClients()
+	if err != nil {
+		return []error{fmt.Errorf("unable to load clients: %s", err)}
+	}
+
+	errs := []error{}
+	for name, client := range clients {
+		// MinCertLifetime, if not set in the config, will default to zero.
+		// In that case this check will still work but only alert if the
+		// certificate is *already* expired.
+		if err := checkCertificate(name, &client, config.MinCertLifetime); err != nil {
+			errs = append(errs, err)
+		}
+
+		// Check that each client has at least one secret. It makes no
+		// sense to have a client without secrets, so if there's an empty
+		// client dir something is probably wrong.
+		if err := checkHasSecrets(name, &client, config.SecretsDir); err != nil {
+			errs = append(errs, err)
+		}
+
+	}
+
+	return errs
+}
+
+func checkCertificate(name string, client *keysync.ClientConfig, minCertLifetime time.Duration) error {
```

Comment:
there should be a test for this (especially for correctly catching expired certs)